### DISCUSSION TITLE
Using function key commands in bash scrip

### DISCUSSION THREAD

**Discussion by gillette7** [GLVis/glvis#202] [Q&A] [answered]
I'm generating some figures using GLVis that requires the use of the "function key" options, namely, F7, for setting the min and max of the color bar.  I can do this interactively using the terminal without issue but is there a way to pass this as part of a bash command?  Thanks!

**Comment by v-dobrev** [comment, marked as answer]
There are two options:
1. generate and run a GLVis script (see https://glvis.org/options-and-use/#glvis-scripts); the equivalent of F7 is the command `valuerange <min> <max>`.
2. create GLVis windows through MFEM sockets from a "C++ script"; this way you can send commands through the socket, in particular, `valuerange <min> <max>` works that way too.

Let me know if you need more details on either of the options.

**Comment by tzanio** [comment]
It is hacky, but you can also provide the input directly from the command line.

The function keys are encoded in the `-k` string as`~6` for `F6`, `~7` for `F7`, etc. If you then create a text file for the input:

```
> cat input.txt

1
1
0
```

you can do something like

```
glvis -m ../mfem/data/escher.mesh -k "~6" < input.txt
```
